File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):
    visited = []
    q=deque()
    cnt = 0
    
    for i in range(n):
        if i not in visited:
            q.append(i)
            cnt+=1
            while q:
                a = q.popleft()
                for i in range(n):
                    if i not in visited and computers[a][i] == 1:
                        visited.append(i)
                        q.append(i)
    return cnt       

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let graph = {}
    let cnt = 0
    for(let i=0;i<computers.length;i++){
        graph[i] = []
        for(let j=0;j<computers[i].length;j++){
            if(i===j) continue
            if(computers[i][j]===1) graph[i].push(j)
        }
    }
    console.log(graph)
    let visited = Array(n).fill(false)
    const dfs = (idx) => {
        visited[idx] = true
        let next = graph[idx]
        for(let i=0;i<next.length;i++){
            if(!visited[next[i]]) dfs(next[i]);
        }
    }
    for(let i=0;i<n;i++){
        if(!visited[i]) {
            dfs(i)
            cnt++;
        }
    }
    return cnt
}